apps/docs/security/delete-protection.mdx

```diff
@@ -0,0 +1,85 @@
+---
+title: Delete Protection
+description: Prevents an API from being deleted when enabled.
```

Comment:
in general: we have delete protection for more than just APIs (right now workspaces and apis) but it'll become more common in the future
let's update these docs to be generic, but we can have examples of enabling it for specific resources like you did below.